### PR TITLE
Update task run status with results

### PR DESCRIPTION
**Pull request by othomann**
# Changes

Corresponding changes to update the task run status with the new results from the task spec.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Provide the task run status update to reflect the new results defined in the task spec.
```

### CODE REVIEW

**Comment by othomann** [comment]
@sbwsg you want the doc changes as part of this PR?

**Comment by sbwsg** [comment]
yes please!

File: docs/tasks.md [hunk 1]
```diff
@@ -356,6 +358,41 @@ steps:
...
+  name: print-date
+  annotations:
+    description: |
+      A simple task that prints the date
+spec:
+  results:
+    - name: current-date-unix-timestamp
+      description: The current date in unix timestamp format
+    - name: current-date-human-readable
+      description: The current date in humand readable format
```
Comment (sbwsg) [inline docs/tasks.md]: `humand` -> `human`

File: docs/tasks.md [hunk 2]
```diff
@@ -502,17 +539,17 @@ is configurable using a flag of the Tekton controller. If the configured "nop"
 image contains the command that the sidecar was running before the sidecar
 was stopped then the sidecar will actually keep running, causing the TaskRun's
 Pod to remain running, and eventually causing the TaskRun to timeout rather
-then exit successfully. Issue https://github.com/tektoncd/pipeline/issues/1347
+then exit successfully. Issue <https://github.com/tektoncd/pipeline/issues/1347>
```
Comment (sbwsg) [inline docs/tasks.md]: Suggest `[Issue 1347](https://github.com/tektoncd/pipeline/issues/1347) has been created to track this bug.`

File: docs/tasks.md [hunk 3]
```diff
@@ -502,17 +539,17 @@ is configurable using a flag of the Tekton controller. If the configured "nop"
...
 ### Variable Substitution
 
 `Tasks` support string replacement using values from:
 
-* [Inputs and Outputs](#input-and-output-substitution)
-  * [Array params](#variable-substitution-with-parameters-of-type-array)
-* [`workspaces`](#variable-substitution-with-workspaces)
-* [`volumes`](#variable-substitution-with-volumes)
+- [Inputs and Outputs](#input-and-output-substitution)
+- [Array params](#variable-substitution-with-parameters-of-type-array)
```
Comment (sbwsg) [inline docs/tasks.md]: I *think* this one is supposed to be nested.  Looking for the heading below it appears to be one level smaller than `Inputs and Outputs`

File: pkg/apis/pipeline/v1alpha1/taskrun_types.go [hunk 4]
```diff
@@ -136,11 +136,24 @@ type TaskRunStatusFields struct {
 	// optional
 	ResourcesResult []PipelineResourceResult `json:"resourcesResult,omitempty"`
 
+	// TaskRunResult from task runs
+	// optional
```
Comment (sbwsg) [inline pkg/apis/pipeline/v1alpha1/taskrun_types.go]: This should be `// +optional` i think.

Comment (sbwsg) [reply pkg/apis/pipeline/v1alpha1/taskrun_types.go]: oh haha, i see that ResourceResult has this wrong as well.

File: pkg/apis/pipeline/v1alpha1/taskrun_types.go [hunk 5]
```diff
@@ -136,11 +136,24 @@ type TaskRunStatusFields struct {
 	// optional
 	ResourcesResult []PipelineResourceResult `json:"resourcesResult,omitempty"`
 
+	// TaskRunResult from task runs
```
Comment (sbwsg) [inline pkg/apis/pipeline/v1alpha1/taskrun_types.go]: `TaskRunResults` to match the field name.  Maybe this should be something like, "TaskRunResults are the list of results written out by the task's containers"

File: docs/taskruns.md [hunk 6]
```diff
@@ -343,6 +344,26 @@ If multiple `steps` are defined in the `Task` invoked by the `TaskRun`, we will
 `spec.steps` of the `Task`, when the `TaskRun` is accessed by the `get` command, e.g.
 `kubectl get taskrun <name> -o yaml`. Replace \<name\> with the name of the `TaskRun`.
 
+### Results
+
+If one or more `results` are defined in the `Task` invoked by the `TaskRun`, we will get an new entry
```
[the remark this answers is not in the document]
Comment (othomann) [reply docs/taskruns.md]: Fixed.

File: docs/tasks.md (hunk 3 above)
Comment (othomann) [reply docs/tasks.md]: I don't see what you mean there. My editor reports issues if '*' are used as being inconsistent with other unordered list from the same document.

Comment (othomann) [reply docs/tasks.md]: Adding indentation to fix this.

**Pull request merged**